### ISSUE TITLE
Rounding for a duration data type

### ISSUE THREAD

**Issue by blaylockbk** [pola-rs/polars#12417]
### Description

In Pandas, you can round Timedeltas,

```python
import pandas as pd
pd.Timedelta("1h20min").round("1h")
```

I expected a similar feature in Polars, but `dt.round` is limited to `date/datetime` types.

```python
import polars as pl
from datetime import datetime, timedelta

a = pl.DataFrame({"times": [timedelta(seconds=15), timedelta(seconds=50)]})
a.select(pl.col("times").dt.round("1m"))
```

> InvalidOperationError: round operation not supported for dtype duration[ns] (expected: date/datetime)

Of course, this can be done [manually](https://stackoverflow.com/q/77456817/2383070), but it would be convenient if Polars could round a Duration with `dt.round`.

Thanks for the consideration.

**Comment by rob-sil** [comment]
Happy to take a swing at this if no one else is!

**Comment by MarcoGorelli** [comment]
yes, sounds like a good feature request if you'd like to try it!
